drivers/periph_common/timer.c

```diff
@@ -19,10 +19,15 @@
  */
 
 #include "periph/timer.h"
+#include "irq.h"
 
 #ifndef PERIPH_TIMER_PROVIDES_SET
 int timer_set(tim_t dev, int channel, unsigned int timeout)
 {
-    return timer_set_absolute(dev, channel, timer_read(dev) + timeout);
+    unsigned int state = irq_disable();
+    unsigned int base = timer_read(dev);
```

Comment:
Is it necessary to use an additional local variable to read the current timer value when no further processing of the current time is taking place, such as testing for underflows?

It's just a question, the compiler will probably generate the same code and keep the current timer value in a register and not require additional stack space.

Reply:
Yeah I think the compiler will be smarter than me here. There is absolutely no reason to not make it explicitly without this variable. Updated it and squashed directly, thanks!

Reply:
Hah, that was a style nit I chose not to mention. :wink: I also like it much better now.